### PR TITLE
Update Unused and __exit__ parameters in stdlib

### PR DESCRIPTION
**Pull request by Avasam**
Follow up to #9475

- Use of `Unused` type alias for unused params.
- Removed an allowlist entry `uuid.getnode` by correctly typing an undocumented/unused param
- Properly(?) typed `asyncio.runners.Runner.__exit__` and `sqlite3.dbapi2.Blob.__exit__` using `type[BaseException] | None, BaseException | None, TracebackType | None` instead of objects. I couldn't check the source for these two, so I'm not sure if the params are meant to be unsued.

There's also 4 `object | None` in stdlib, should I change those to `object` as well?

### CODE REVIEW

**Comment by Avasam** [comment]
Random CI failure, you can re-run just that job

**Comment by AlexWaygood** [comment]
Hmm, while support for _arbitrary_ aliases in flake8-pyi is unrealistic, we could certainly add some special-casing for the `_typeshed.Unused` alias specifically. I think I'd prefer to do that rather than add a bajillion `# noqa` comments :)

**Comment by AlexWaygood** [comment]
> Hmm, while support for _arbitrary_ aliases in flake8-pyi is unrealistic, we could certainly add some special-casing for the `_typeshed.Unused` alias specifically. I think I'd prefer to do that rather than add a bajillion `# noqa` comments :)

Are you interested in working on a flake8-pyi PR for this? If not, I can take a look at adding special-casing

**Comment by Avasam** [comment]
> Are you interested in working on a flake8-pyi PR for this? If not, I can take a look at adding special-casing

Sure. If I"m stuck I"ll let you know

File: stdlib/sqlite3/dbapi2.pyi [hunk 1]
```diff
@@ -227,7 +227,7 @@ else:
 
 if sys.version_info < (3, 8):
     class Cache:
-        def __init__(self, *args: Incomplete, **kwargs: object) -> None: ...
+        def __init__(self, *args: Incomplete, **kwargs: Unused) -> None: ...
```
Comment (AlexWaygood) [inline stdlib/sqlite3/dbapi2.pyi:230]: I'm not sure the existing annotation for this is correct, but also: it's 3.7-only, so I'm not sure I care too much 😛

File: stdlib/sqlite3/dbapi2.pyi [hunk 2]
```diff
@@ -453,6 +453,8 @@ if sys.version_info >= (3, 11):
         def seek(self, __offset: int, __origin: int = ...) -> None: ...
         def __len__(self) -> int: ...
         def __enter__(self: Self) -> Self: ...
-        def __exit__(self, __typ: object, __val: object, __tb: object) -> Literal[False]: ...
+        def __exit__(
+            self, __typ: type[BaseException] | None, __val: BaseException | None, __tb: TracebackType | None
+        ) -> Literal[False]: ...
```
Comment (AlexWaygood) [inline stdlib/sqlite3/dbapi2.pyi:458]: This signature was previously discussed in https://github.com/python/typeshed/pull/7625#discussion_r851630434: the parameters here really do take any object

**Review by AlexWaygood — COMMENTED** [review]
Mostly looks good, but a few points below! Also, you have a fair few unused `noqa` comments following the new flake8-pyi release ;)

File: stdlib/sqlite3/dbapi2.pyi (hunk 1 above)
Comment (Avasam) [reply stdlib/sqlite3/dbapi2.pyi:230]: Nor can I find "Cache" in the docs 🤷 
https://docs.python.org/3.7/library/sqlite3.html

**Comment by Avasam** [comment]
Thanks for looking into the methods I was uncertain about.

I still have one unanswered question left:
> There's also 4 `object | None` in stdlib, should I change those to object as well?

**Comment by AlexWaygood** [comment]
> I still have one unanswered question left:
> 
> > There's also 4 `object | None` in stdlib, should I change those to `object` as well?

Sure, why not!

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**